showkase-processor/src/main/java/com/airbnb/android/showkase/processor/writer/ShowkaseMethodsWriter.kt

```diff
@@ -0,0 +1,154 @@
+                    .addFunction(
+                        generateMetadataFunction(
+                            rootElement, 
+                            "$rootModulePackageName.$rootModuleClassName"
+                        )
+                    )
+                    .build()
+            )
+            .build()
+            .writeTo(processingEnv.filer)
+    }
+
+    private fun generateIntentFunction(
+        rootModulePackageName: String,
+        rootModuleClassName: String,
+        rootElement: Element,
+    ) = FunSpec.builder(INTENT_FUNCTION_NAME).apply {
+        addParameter(
+            CONTEXT_PARAMETER_NAME, CONTEXT_CLASS_NAME
+        )
+        returns(INTENT_CLASS_NAME)
+        addCode(
+            CodeBlock.Builder()
+                .indent()
+                .addStatement(
+                    "val intent = %T(%L, %T::class.java)",
+                    INTENT_CLASS_NAME,
+                    CONTEXT_PARAMETER_NAME,
+                    SHOWKASE_BROWSER_ACTIVITY_CLASS_NAME
+                )
+                .addStatement(
+                    "intent.putExtra(%S, %S)",
+                    SHOWKASE_ROOT_MODULE_KEY,
+                    "$rootModulePackageName.$rootModuleClassName"
+                )
+                .addStatement(
+                    "return intent"
+                )
+                .unindent()
+                .build()
+        )
+        addOriginatingElement(rootElement)
+    }
+        .build()
+
+    private fun generateMetadataFunction(
+        rootElement: Element,
+        classKey: String
+    ) = FunSpec.builder(METADATA_FUNCTION_NAME).apply {
+        returns(SHOWKASE_ELEMENTS_METADATA_CLASS_NAME)
+        addCode(
+            CodeBlock.Builder()
+                .indent()
+                .addStatement("try {")
+                .indent()
+                .addStatement(
+                    "val showkaseComponentProvider = Class.forName(\"${classKey}Codegen\").newInstance() as %T",
+                    SHOWKASE_PROVIDER_CLASS_NAME
+                )
+                .generateShowkaseElementProperty(
+                    "componentsMap", 
+                    "getShowkaseComponents", 
+                    "group"
+                )
+                .generateShowkaseElementProperty(
+                    "colorsMap", 
+                    "getShowkaseColors", 
+                    "colorGroup"
+                )
+                .generateShowkaseElementProperty(
+                    "typographyMap", 
+                    "getShowkaseTypography", 
+                    "typographyGroup"
+                )
+                .addStatement(
+                    "return %T(%L, %L, %L)",
+                    SHOWKASE_ELEMENTS_METADATA_CLASS_NAME,
+                    "componentsMap",
+                    "colorsMap",
+                    "typographyMap"
+                )
+                .unindent()
+                .addStatement("} catch(exception: ClassNotFoundException) {")
```

Comment:
Two questions: Why would a CNF exception happen? And is an metadata holder the right default?

Reply:
I think notifying the user is a reasonable thing to do for this function.